wandb/sdk/launch/agent/agent.py

```diff
@@ -421,6 +426,30 @@ async def finish_thread_id(
         """Removes the job from our list for now."""
         with self._jobs_lock:
             job_and_run_status = self._jobs[thread_id]
+
+        try:
```

Comment:
Actually thinking about this more, it would make more sense to have the `KubernetesRunner` create the additional services, attach them to the `KubernetesSubmittedRun` and add a cleanup method in the KubernetesSubmittedRun